### ISSUE TITLE
scapy.utils.RawPcapReader.read_packet() in scapy's v2.4 is incompatible against scapy v2.3.2 & 2.3.3

### ISSUE THREAD

**Issue by karpierz** [secdev/scapy#1128]
Scapy Version: 2.4.0rc4-58
System: Windows10
Python Version: 2.7.14

Scapy 2.4 has incompatibility with in scapy.utils.RawPcapReader.read_packet() against v2.3.2 & 2.3.3
Previously returned value was:
```return s,(sec,usec,wirelen) # caplen = len(s)```
but in scapy 2.4 is:
```return s, (sec, usec, caplen, wirelen)```

Its breaks a lot of scripts in our customers :(.
Unfortunately full backward compatibility is impossible without omitting 'caplen'
but in 99% our cases restoring the order of tuple items will be quite enough. 
Additionally using named tuple instead of tuple will allow to avoid any other incompatibilities
with more elegant/proper way.

Proposal below (need some investigations/discussion with scapy authors/maintainers due to
possible impact for inner pieces of new scapy):
PS: If patch will be accepted, I of course will make appropriate pull request.

```
diff --git a/scapy/utils.py b/scapy/utils.py
index dde2f4e..7dfd4c2 100644
--- a/scapy/utils.py
+++ b/scapy/utils.py
@@ -15,6 +15,7 @@ import gzip, zlib
 import re, struct, array
 import subprocess
 import tempfile
+import collections
 
 import warnings
 import scapy.modules.six as six
@@ -816,6 +817,8 @@ class RawPcapReader(six.with_metaclass(PcapReader_metaclass)):
         return pkt
     __next__ = next
 
+    _PacketInfo = collections.namedtuple("PacketInfo",
+                                         ["sec", "usec", "wirelen", "caplen"])
 
     def read_packet(self, size=MTU):
         """return a single packet read from the file
@@ -827,7 +830,7 @@ class RawPcapReader(six.with_metaclass(PcapReader_metaclass)):
             return None
         sec,usec,caplen,wirelen = struct.unpack(self.endian+"IIII", hdr)
         s = self.f.read(caplen)[:size]
-        return s, (sec, usec, caplen, wirelen)
+        return s, self._PacketInfo(sec=sec, usec=usec, wirelen=wirelen, caplen=caplen)
 
 
     def dispatch(self, callback):
@@ -882,7 +885,7 @@ class PcapReader(RawPcapReader):
         rp = super(PcapReader, self).read_packet(size=size)
         if rp is None:
             return None
-        s, (sec, usec, caplen, wirelen) = rp
+        s, pkt_info = rp
         
         try:
             p = self.LLcls(s)
@@ -892,8 +895,8 @@ class PcapReader(RawPcapReader):
             if conf.debug_dissector:
                 raise
             p = conf.raw_layer(s)
-        p.time = sec + (0.000000001 if self.nano else 0.000001) * usec
-        p.wirelen = wirelen
+        p.time = pkt_info.sec + (0.000000001 if self.nano else 0.000001) * pkt_info.usec
+        p.wirelen = pkt_info.wirelen
         return p
     def read_all(self,count=-1):
         res = RawPcapReader.read_all(self, count)
```

Thanks,
Adam Karpierz

**Comment by p-l-** [comment]
Sounds OK to me, I think you should submit a PR.

**Comment by karpierz** [comment]
PR has completed successfully.
Branch is ready for merge to the master.

Thanks,
Adam Karpierz

**Comment by p-l-** [comment]
Fixed in #1133.
